tests/gpu/main_code.cpp

```diff
@@ -0,0 +1,37 @@
+// clang-format off
+// RUN: ./rdc.%ext | FileCheck %s --check-prefixes=CHECK,CHECK-FIRST
+// Second run uses the object cache.
+// RUN: ./rdc.%ext | FileCheck %s --check-prefixes=CHECK,CHECK-SECOND
```

Comment:
Naming for the run program seems off. It should be the executable plus the lit-provided extensions. I guess `main_code.%ext`